### PR TITLE
Kata train reservation 1

### LINKED ISSUE #1: Make sure train data service test still passes
Make sure `train_data_service/test_train_data_service.py` still passes as the service now expects a reservation JSON object as request body.

### PR DESCRIPTION
**Pull request by mrclneumann**
Add functional tests on REST API layer

Run with:
```bash
pytest train_data_service/test_train_data_service.py
```

Closes #1

### CODE REVIEW

File: train_data_service/train_data_service.py [hunk 1]
```diff
@@ -40,9 +44,10 @@ def reset(self, train_id):
             seat["booking_reference"] = ""
         return self.data_for_train(train_id)
 
+
 def start(trains_data):
     TrainDataService.data_for_train.exposed = True
```
Comment (stefanscheidt) [inline train_data_service/train_data_service.py:49]: I don't know Cherrypy but as we now have the `@cherrypy.expose()` annotations, do we still need those `TrainDataService.xxx.exposed = True` stuff?

**Review by stefanscheidt — CHANGES_REQUESTED** [review]
Do we need `__init__.py` in the repo or should it be ignored?

File: train_data_service/requirements.txt [hunk 2]
```diff
@@ -0,0 +1,26 @@
+astroid==2.4.2
```
Comment (stefanscheidt) [inline train_data_service/requirements.txt:1]: How do we have to use this `requirements.txt`? Do I have to do a `pip3 install -r requirements.txt` at the beginning? Probably something we should add to the README?

**Review by stefanscheidt — CHANGES_REQUESTED** [review]

File: train_data_service/train_data_service.py (hunk 1 above)
Comment (mrclneumann) [reply train_data_service/train_data_service.py:49]: Indeed, this extra setup is redundant now. Will be removed.

**Review by stefanscheidt — APPROVED** [review]

**Pull request merged**